packages/backend/src/server/api/endpoints/admin/ad/create.ts

```diff
@@ -10,27 +9,17 @@ export const meta = {
 	requireModerator: true,
 
 	params: {
-		url: {
-			validator: $.str.min(1),
-		},
-		memo: {
-			validator: $.str,
-		},
-		place: {
-			validator: $.str,
-		},
-		priority: {
-			validator: $.str,
-		},
-		ratio: {
-			validator: $.num.int().min(0),
-		},
-		expiresAt: {
-			validator: $.num.int(),
-		},
-		imageUrl: {
-			validator: $.str.min(1),
+		type: 'object',
+		properties: {
+			url: { type: 'string', minLength: 1, },
+			memo: { type: 'string', },
+			place: { type: 'string', },
+			priority: { type: 'string', },
+			ratio: { type: 'integer', },
```

Comment:
Is the minimum value no longer needed? I think it would stil be necessary to check that it is not negative.
```suggestion
			ratio: { type: 'integer', minimum: 0, },
```
This is only one example, minimum values were not transferred in more places.